direct.c

```diff
@@ -44,8 +44,13 @@ static int
 direct_listxattr (struct ovl_layer *l, const char *path, char *buf, size_t size)
 {
   char full_path[PATH_MAX];
-
-  strconcat3 (full_path, PATH_MAX, l->path, "/", path);
+  int ret;
+  ret = snprintf (full_path, sizeof (full_path), "/proc/self/fd/%d/%s", l->fd, path);
+  if (ret == PATH_MAX)
```

Comment:
I am not sure but should this be `if (ret >= PATH_MAX)` ?

Reply:
Probably should be sizeof(full_path)

Reply:
yes, thanks.  Fixed to `>= sizeof (full_path)`